phases/ephemeral/witx/typenames.witx

```diff
@@ -5,735 +5,735 @@
-(typename $event_fd_readwrite_t
+(typename $event_fd_readwrite
   (struct
     ;; The number of bytes available for reading or writing.
-    (field $nbytes $filesize_t)
+    (field $nbytes $filesize)
     ;; The state of the file descriptor.
-    (field $flags $eventrwflags_t)
+    (field $flags $eventrwflags)
   )
 )
 
-;; The contents of an $event_t.
+;; The contents of an $event.
 (typename $event_u
   (union
     ;; When type is `EVENTTYPE_FD_READ` or `EVENTTYPE_FD_WRITE`:
-    (field $fd_readwrite $event_fd_readwrite_t)
+    (field $fd_readwrite $event_fd_readwrite)
   )
 )
 
 ;; An event that occurred.
-(typename $event_t
+(typename $event
   (struct
     ;; User-provided value that got attached to `subscription_t::userdata`.
-    (field $userdata $userdata_t)
+    (field $userdata $userdata)
     ;; If non-zero, an error that occurred while processing the subscription request.
-    (field $error $errno_t)
+    (field $error $errno)
     ;; The type of the event that occurred.
-    (field $type $eventtype_t)
+    (field $type $eventtype)
     ;; The contents of the event.
     (field $u $event_u)
   )
 )
 
 ;; Flags determining how to interpret the timestamp provided in
 ;; `subscription_t::u.clock.timeout.`
-(typename $subclockflags_t
+(typename $subclockflags
   (flags u16
     ;; If set, treat the timestamp provided in
     ;; `subscription_t::u.clock.timeout` as an absolute timestamp of clock
     ;; `subscription_t::u.clock.clock_id.` If clear, treat the timestamp
     ;; provided in `subscription_t::u.clock.timeout` relative to the
     ;; current time value of clock `subscription_t::u.clock.clock_id.`
-    $SUBSCRIPTION_CLOCK_ABSTIME
+    $clock_abstime
   )
 )
 
-;; The contents of a $subscription_t when type is `EVENTTYPE_CLOCK`.
-(typename $subscription_clock_t
+;; The contents of a $subscription when type is `EVENTTYPE_CLOCK`.
+(typename $clock
   (struct
     ;; The clock against which to compare the timestamp.
-    (field $clock_id $clockid_t)
+    (field $id $clockid)
     ;; The absolute or relative timestamp.
-    (field $timeout $timestamp_t)
+    (field $timeout $timestamp)
     ;; The amount of time that the implementation may wait additionally
     ;; to coalesce with other events.
-    (field $precision $timestamp_t)
+    (field $precision $timestamp)
     ;; Flags specifying whether the timeout is absolute or relative
-    (field $flags $subclockflags_t)
+    (field $flags $subclockflags)
   )
 )
 
-;; The contents of a $subscription_t when type is type is
+;; The contents of a $subscription when type is type is
 ;; `EVENTTYPE_FD_READ` or `EVENTTYPE_FD_WRITE`.
-(typename $subscription_fd_readwrite_t
+(typename $fd_readwrite
   (struct
     ;; The file descriptor on which to wait for it to become ready for reading or writing.
-    (field $file_descriptor $fd_t)
+    (field $file_descriptor $fd)
   )
 )
 
-;; The contents of a $subscription_t.
-(typename $subscription_u
+;; The contents of a $subscription.
+(typename $u
```

Comment:
This ended up as a type named `$u` at the top-level, which would be good to avoid.